aten/src/ATen/native/QuantizedLinear.cpp

```diff
@@ -40,6 +166,10 @@ Tensor fbgemm_linear_int8_weight(
     Scalar weight_scale,
     Scalar weight_zero_point,
     const Tensor& bias) {
+  TORCH_WARN(
+      "fbgemm_linear_int8_weight will be deprecated soon."
+      "Please use fbgemm_linear_int8_weight_fp32_activation instead.");
+
   // We make a strong guarantee that models using these operators will have the
```

Comment:
From here, the implementation is same as fbgemm_linear_int8_weight_fp32_activation . Maybe we can just call that function after printing deprecation warning? (I think this is what James suggested)